pkg/compactor/shard_aware_deduplicate_filter.go

```diff
@@ -227,14 +236,14 @@ func (b *blockWithSuccessors) addSuccessorIfPossible(other *blockWithSuccessors)
 	return true
 }
 
-// isFullyIncludedInSuccessors returns true if this block is *fully* included in successor blocks.
-// This is true under following conditions:
+// isFullyIncludedInSuccessors returns true if this block is *fully* included in its own successor blocks.
+// This is true under the following conditions:
 //
-// - if this block has non-empty shardID and has *any* successors, then it is fully included already.
+// - if this block has a non-empty shardID, and it has *any* successors, then the block is fully included in its successors.
```

Comment:
I think "it is" makes sense:
```suggestion
// - if this block has a non-empty shardID, and it has *any* successors, then it is fully included in its successors.
```